package/piksi_leds/src/led_adp8866.c

```diff
@@ -215,12 +219,19 @@ static bool leds_set(const led_adp8866_led_state_t *led_states,
 
       /* Write ISCn */
       if (i2c_write(isc_route[led_state->led],
-                    (led_state->brightness << LED_ADP8866_ISCn_SCDn_Pos)) !=
-          0) {
-        ret = false;
-      } else {
+                    (led_state->brightness << LED_ADP8866_ISCn_SCDn_Pos))
+          != 0) {
+        if (errno == ETIMEDOUT) {
+          ret = (i2c_write(isc_route[led_state->led],
+                     (led_state->brightness << LED_ADP8866_ISCn_SCDn_Pos)) == 0);
+        }
+      }
+
+      if (ret) {
         /* Update cache */
         brightness_cache[led_state->led] = led_state->brightness;
+      } else {
```

Comment:
I know this is WIP, but if this fixes the issue we should probably add a error logging & a cache flush here-- maybe add a retry loop here for a max of 3 retries?